### PR TITLE
Test updating dev-requirements.txt / setup.py on workflow dispatch

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
Install core dependencies for integration testing from dbt-core branch.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by colin-rogers-dbt** [comment]
Manually kicked off test run: https://github.com/dbt-labs/dbt-redshift/actions/runs/4235851680/jobs/7359978550

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -34,6 +34,11 @@ on:
   # manual trigger
   workflow_dispatch:
   # run this once per night to ensure no regressions from latest dbt-core changes
+    inputs:
+      dbt-core-branch:
+        description: "branch of dbt-core to use in dev-requirements.txt"
+        required: false
+        type: string
   schedule:
```
Comment (emmyoop) [inline .github/workflows/integration.yml:42]:
```suggestion
  workflow_dispatch:
    inputs:
      dbt-core-branch:
        description: "branch of dbt-core to use in dev-requirements.txt"
        required: false
        type: string
  # run this once per night to ensure no regressions from latest dbt-core changes
  schedule:
```

**Review by emmyoop — CHANGES_REQUESTED** [review]
Won't this also require always having a branch in `dev-requirements.txt`, even if it's just `main`?

**Comment by colin-rogers-dbt** [comment]
> Won't this also require always having a branch in `dev-requirements.txt`, even if it's just `main`?

Actually putting a branch in dev-requirements would break this since I just prepend branch name i.e. you would end up with:
`git+https://github.com/dbt-labs/dbt-core.git@main@main#egg=dbt-core&subdirectory=core`

**Comment by emmyoop** [comment]
But then we can never run this for workflows that have a base of one of the latest branches.  Are we okay with that?  It's not common, but not unheard of either.

**Review by emmyoop — APPROVED** [review]
Approving knowing this may not work for windows.  Since it can only be invoked on purpose nothing current will break.  You can't actually test this until it's in main anyways,

**Comment by colin-rogers-dbt** [comment]
> But then we can never run this for workflows that have a base of one of the latest branches. Are we okay with that? It's not common, but not unheard of either.

Good call out, update the sed syntax to handle this! 

> Approving knowing this may not work for windows. 

Looks like it does run on Windows! 

> You can't actually test this until it's in main anyways.
Since this is an update to an existing workflow I am able to test it off a branch with:
`gh workflow run integration.yml -f dbt-core-branch=cl/dbt-retry --ref update_dev_req`
See: https://github.com/dbt-labs/dbt-redshift/actions/runs/4237292878/jobs/7363103011

**Pull request merged**